.github/workflows/integration-tests.yml

```diff
@@ -0,0 +1,59 @@
+name: Integration Tests
+
+on:
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  build_and_test_triton_shared:
+    runs-on: ubuntu-latest
+
+    steps:
+
+    - name: Checkout Triton
+      uses: actions/checkout@v4
+      with:
+        repository: 'openai/triton'
+        ref: '215b2e77a1d92f907bc4addfa3c3ba204028c32e' # known good commit "Add Shared Middle Layer to Triton via Plug-In"
+        path: triton
+        submodules: 'recursive'
```

Comment:
nit: This will checkout all the other submodules as well, which we don't need.

edit: especially since you are cloning triton-shared directly into it's third-party directory, so we don't need any submodules at all

Reply:
You're absolutely correct, but I'm going to leave this in place. The check-out takes all of 12s and this future proofs us in case we want other submodules.